### ISSUE TITLE
[BUG]  master-0bde441-head cannot build storage classes

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#6631]
**Describe the bug**
Cannot build a storage class, error is:
> admission webhook "validator.harvesterhci.io" denied the request: storage class base is not for encryption or decryption

**To Reproduce**
1. Modify some hosts, add extra storage
1. Build a storage class either via Form or Edit As YAML, that targets the nodes new force-formatted & tagged storage
1. It will fail 

**Expected behavior**
The ability to add the storage class

**Workaround**
The workaround is to remove the "encrypted" property from the YAML in the `parameters` section:
```
__clone: true
metadata:
  annotations:
    field.cattle.io/description: base storage class to target secondary volumes
  name: base
  labels: {}
parameters:
  numberOfReplicas: '3'
  staleReplicaTimeout: '30'
  diskSelector: extra
  nodeSelector: hosts
  migratable: 'true'
provisioner: driver.longhorn.io
allowVolumeExpansion: true
reclaimPolicy: Delete
volumeBindingMode: Immediate
apiVersion: storage.k8s.io/v1
kind: StorageClass
```

**Other Attempted Workarounds, that didn't workout, so can ignore**
1. trying to change to `encrypted: false` -> will yield:  `StorageClass in version "v1" cannot be handled as a StorageClass: json: cannot unmarshal bool into Go struct field StorageClass.parameters of type string`
2. trying to just set it to `encrypted: 'true'` -> will yield: `admission webhook "validator.harvesterhci.io" denied the request: storage class must contain csi.storage.k8s.io/provisioner-secret-name, csi.storage.k8s.io/provisioner-secret-namespace, csi.storage.k8s.io/node-stage-secret-name, csi.storage.k8s.io/node-stage-secret-namespace, csi.storage.k8s.io/node-publish-secret-name, csi.storage.k8s.io/node-publish-secret-namespace`

**Environment**
 - Harvester ISO version: master-0bde441-head
 - Underlying Infrastructure: 4 node bare-metal hp dl160 gen9s

**Additional context**
![Screenshot from 2024-09-24 16-33-54](https://github.com/user-attachments/assets/d58f593f-cfef-4207-895e-b83b9517a30e)
![Screenshot from 2024-09-24 16-32-10](https://github.com/user-attachments/assets/0ef1f100-af01-478b-aa50-a99ac5e54eb4)
![Screenshot from 2024-09-24 16-29-20](https://github.com/user-attachments/assets/b872c740-5bb0-45b4-81a9-e6a567c6830e)

**Comment by Yu-Jack** [comment]
> trying to change to encrypted: false -> will yield: StorageClass in version "v1" cannot be handled as a StorageClass: json: cannot unmarshal bool into Go struct field StorageClass.parameters of type string

The reason is that the `spec.Parameters` in StorageClass is `map[string]string` structure. So, it's expected behavior and error If we change the value to bool type.

**Comment by Yu-Jack** [comment]
Hi @irishgordo, I just tested with [v1.4-4947368](https://github.com/harvester/harvester/commit/49473689754483f4a6233dd6c8546a6ef2c852d2)  and [master-c3ab1b6](https://github.com/harvester/harvester/commit/c3ab1b673e84f164367bf7264264758cdd978883). Recordings are below.


https://github.com/user-attachments/assets/7be915f9-ba0b-44a4-982f-c028630af486


https://github.com/user-attachments/assets/f0d1dc43-80b1-4e47-b659-4c4414ed387b

**Comment by irishgordo** [comment]
@Yu-Jack thanks for that mention :smile: :+1:  - since this has now been fixed in the heads of master & v1.4 I'll go ahead and close this out :+1:
